### PR TITLE
Add the `login-unknown` configuration option

### PR DESCRIPTION
**Pull request by wolneykien**
If the option is set to `True`, LightDM proceeds to start the new session even if the username is not known. Intended to be used with `pam_mkuser.so` PAM module.

### CODE REVIEW

File: src/seat.c [hunk 1]
```diff
@@ -998,8 +998,15 @@ create_user_session (Seat *seat, const gchar *username, gboolean autostart)
     l_debug (seat, "Creating user session");
 
     /* Load user preferences */
+    gboolean login_unknown = FALSE;
+
+    if (seat_get_string_property (seat, "login-unknown"))
```
Comment (robert-ancell) [inline src/seat.c:1003]: Please add this into `data/lightdm.conf` with the default value and some documentation.

File: src/seat.c [hunk 2]
```diff
@@ -998,8 +998,15 @@ create_user_session (Seat *seat, const gchar *username, gboolean autostart)
     l_debug (seat, "Creating user session");
 
     /* Load user preferences */
+    gboolean login_unknown = FALSE;
+
```
Comment (robert-ancell) [inline src/seat.c:1002]: Nit: remove empty line.

File: src/seat.c [hunk 3]
```diff
@@ -998,8 +998,15 @@ create_user_session (Seat *seat, const gchar *username, gboolean autostart)
...
+    if (seat_get_string_property (seat, "login-unknown"))
+        login_unknown = seat_get_boolean_property (seat, "login-unknown");
```
Comment (robert-ancell) [inline src/seat.c:1004]: login-unknown is a bit of a cryptic name, I suggest "allow-empty-username"

**Comment by robert-ancell** [comment]
This should also have a test, see `tests/scripts/login-long-username.conf` for one to copy. (I appreciate the tests are a bit hard to write, so please have a go but if not I'll add them when I get time).

**Comment by robert-ancell** [comment]
Actually should really add two tests (empty username, one with config option and one without), as we weren't testing an empty username.

File: src/seat.c (hunk 3 above)
Comment (robert-ancell) [reply src/seat.c:1004]: Now I look at this again, it is unknown users, not just empty usernames.

**Comment by robert-ancell** [comment]
Now I look at this a second time, I think the correct solution is to always allow this. I can't see a reason for us to fail at this point - it should fail in the PAM layer if the username is incorrect.
